.github/actions/build/action.yml

```diff
@@ -30,11 +50,20 @@ runs:
 
     - name: Build the o1js bindings
       if: steps.bindings-cache.outputs.cache-hit != 'true'
+      env:
+        AWS_ACCESS_KEY_ID: ${{ inputs.cache_id }}
+        AWS_SECRET_ACCESS_KEY: ${{ inputs.gcp_secret }}
       shell: bash
       run: |
         set -Eeu
         ./pin.sh
+        nix build o1js#bindings --accept-flake-config
+        temp_key=$(mktemp)
+        echo ${{ inputs.nar_secret }} > "$temp_key"
+        nix store sign --key-file "$temp_key" --recursive ./result
+        nix copy --to "s3://mina-nix-cache?endpoint=https://storage.googleapis.com" $(nix path-info ./result)
```

Comment:
What is the implication of cahing the nix build?  I assume that users on their local machine will be able to develop locally much faster by downloading freshly cached dependencies?

Reply:
Yeah. We already sort of had this, the difference being made here is that we are adding things built in o1js CI to that cache so the cache should be much more useful than when it was only mina builds.